core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -778,6 +809,18 @@ oppia.directive('conversationSkin', [
           };
 
           $scope.showUpcomingCard = function() {
+            var currentIndex = PlayerPositionService.getActiveCardIndex();
+            if ($scope.activeCard.stateName === null &&
```

Comment:
I feel like the check here ought to be something like "isInConceptCardMode". "stateName === null" doesn't really mean anything conceptually...

Also, I still think it might be nice if activeCard was a proper domain object, and you could call functions on it, e.g. activeCard.isConceptCard(). That would probably simplify things somewhat. What do you think? This should also resolve the comments above about "interaction === null" -- you can use a similar check there.

Reply:
I added an isStateShowingConceptCard() function in ExplorationPlayerStateService. Is that fine? Since, all other checks like whether the state is terminal are all present there.

Reply:
Done, created the function in ExplorationPlayerStateService.